src/components/List/List.jsx

```diff
@@ -64,6 +70,55 @@ const List = () => {
 
   const clearFilter = () => setInputValue('');
 
+  const groups = (list) => {
```

Comment:
Nice work on this! I'm super impressed.

If we wanted to make this this a little more readable we may want to consider changing the data model of this function because an array of arrays feels a little bit difficult to work with. Instead, I would recommend using object and then using `Object.entries` to easily convert it into an array of keys and values to iterate over later on. I'm thinking something that look like:

```javascript
const groupsDictionary = {
  Soon: [],
  'Kind of Soon': [],
  'Not Soon': [],
  'Inactive': [], 
};
```

Alternatively, you could try a slightly different approach. Right now, in order to consume this, we have to use a `.map` within a `.map`, which is a nested loop. Usually nested loops are thought of as inefficient algorithms because the number of loops performed increases exponentially for each item added. However, in this case because the length of the outer loop is always fixed, I don't think that we'd run into that issue. Nevertheless, I usually try as hard as I can to avoid nested-loops because usually they're best to avoid at all costs.

In this case, rather than pushing the values into a group, we could instead begin by using `.sort` in order to sort the groups. Then, when we're mapping over them, we can determine their age grouping dynamically via an algorithm similar to what you have below.

In pseudo-code, it would look like:

`docs.sort((a,b) => ...do the sorting...).map(doc => <li className={`${getAgeGroupClassName(doc)}`}....

Reply:
@MikeBLambert With the second option, the question is about the "inactive" items. 
Because we are already getting the list from firebase sorted by lastEstimated and then by name. But we don't know how to make the inactive elements appear in a different group and not be included in the other groups, since their condition is different.

> Inactive (when there is only one purchase in the database or the purchase is really out of date [the time that has elapsed since the last purchase is twice the estimated])